### ISSUE TITLE
Factory sample fails on PlatformIO build

### ISSUE THREAD

**Issue by hpsaturn** [Xinyuan-LilyGO/T-Display-S3#4]
## Overview

Adding the fix from the PR #3 in this sample, for fix similar issue than #2, I have the next output:

```cpp
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [=         ]  12.2% (used 39972 bytes from 327680 bytes)
Flash: [======Error: The program size (1323221 bytes) is greater than maximum allowed (1310720 bytes)
*** [checkprogsize] Explicit exit, status 1
====]  101.0% (used 1323221 bytes from 1310720 bytes)
```

If I comment this lines:

```python
platform_packages =
    framework-arduinoespressif32@https://github.com/espressif/arduino-esp32.git#2.0.3
```
the compiling disapear, but the factory don't works, and the output is:

```cpp
 ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x42098d1e
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0x39c
load:0x403b6000,len:0x9a4
load:0x403ba000,len:0x2868
SHA-256 comparison failed:
Calculated: 9127e332554622857d5c753c9ffbb9862bfb72e81ded80db80617d5974000cec
Expected: 867c0d6cb8d08f5a22bebcb1e4a5d457c2700a1c6b76abeb4d778eadcc3446e2
Attempting to boot anyway...
entry 0x403b61c0
E (308) psram: PSRAM ID read error: 0x00ffffff
``` 

**Maybe we need arduino-esp32 2.0.3 version but with different partition scheme?**

**Comment by LilyGO** [comment]
It should be that the animation or font takes up a lot of space, resulting in insufficient flash space. In this case, the partition table can be modified to expand the space for storing programs.

**Comment by hpsaturn** [comment]
I tested it with two partitions schemas but I had the same error.

**Comment by LilyGO** [comment]
Maybe there is still a problem with the configuration of platfromIO.

**Comment by hpsaturn** [comment]
I will submit a Github action for try to probe this hypothesis

**Comment by LilyGO** [comment]
I am reconfiguring platfromIO. ini

**Comment by LilyGO** [comment]
I have reconfigured the platfromIO.ini file in the factory

**Comment by hpsaturn** [comment]
From the continuous integration workflow (with my last PR #6 ) sound like that the compiling issue was resolved. I'm out of my house, I don't have here the board, but sounds like that will works.

**Comment by Jason2866** [comment]
Why are you using still the outdated Arduino Core 2.0.3? Many bug fixes have been done in actual Core 2.0.5.
